crates/cargo-test-support/src/publish.rs

```diff
@@ -75,8 +75,11 @@ fn _validate_upload(
     f.read_exact(&mut json_bytes).expect("read JSON data");
     let actual_json = serde_json::from_slice(&json_bytes).expect("uploaded JSON should be valid");
     let expected_json = serde_json::from_str(expected_json).expect("expected JSON does not parse");
-    find_json_mismatch(&expected_json, &actual_json)
-        .expect("uploaded JSON did not match expected JSON");
+
+    if let Err(e) = find_json_mismatch(&expected_json, &actual_json) {
+        println!("{}", e);
+        panic!();
```

Comment:
The `println!` and `panic!` lines can get folded together here I think with `panic!("{}", e)`

Reply:
:sweat_smile: done